kafka/kafka_test.go

```diff
@@ -109,9 +109,12 @@ var _ = Describe("Subscriber", func() {
 		BeforeEach(func() {
 			shared = lint.SubscriberInput{
 				Subject: func(topic string, messages []bps.SubMessage) bps.Subscriber {
-					Expect(seedMessages(topic, messages)).To(Succeed())
```

Comment:
see, it's a bit awkward to have topic/messages passed to both, Subject could be just a value not a func

Reply:
I had to do that for kafka, it's slower and its easier to test it with start=oldest, 'cause otherwise it fails randomly. Not a fan of this either, but it's like:

- usual pubsub (start=newest): seed msgs in Seed
- kafka pubsub (start=oldest): seed msgs in Subject

Will try to revise this, but not sure I'll come up with anything more proper.

Reply:
sure, all I am saying is, do we really need Seed and Subject both accepting topic/messages, it's OK to split the two and one just being a `Subject bps.Subscriber`, the other one being a function and doing the seeding

Reply:
Actually, I can simplify Subject to be just a prop - I can seed messages right in BeforeEach for start=oldest :man_facepalming:

Reply:
Maybe the best way would be to have smth like `SubscriberInput.Behaviour = newest/oldest + .Seed`: then we can run pub/sub in proper order:

```
behaviour=newest:
- init/get subject
- subscribe
- seed

behaviour=oldest:
- seed
- init/get subject
- subscribe
```

For InMem impl that should require some kind of "Replace" method, but for others - should work without any impl modification.

Reply:
Maybe yes: then we can even run lint 2x for stuff that support both behaviours (like STAN) - one for newest, one for oldest to get both covered.

Reply:
Addressed in https://github.com/bsm/bps/pull/31